docs/execution-providers/DirectML-ExecutionProvider.md

```diff
@@ -98,23 +135,40 @@ If creating the onnxruntime InferenceSession object directly, you must set the a
 
-Normally when the shapes of model inputs are known during session creation, the shapes for the rest of the model are inferred by OnnxRuntime when a session is created.  However if a model input contains a free dimension (such as for batch size), steps must be taken to retain the above performance benefits.
+The DirectML execution provider works most efficiently when tensor shapes are known at the time a session is created.  Here are a few performance benefits:
 
-In this case, there are three options:
-- Edit the model to replace an input's free dimension (specified through ONNX using "dim_param") with a fixed size (specified through ONNX using "dim_value").
-- Specify values of named dimensions within model inputs when creating the session using the OnnxRuntime *AddFreeDimensionOverrideByName* ABI.
-- Edit the model to ensure that an input's free dimension has a [denotation](https://github.com/onnx/onnx/blob/master/docs/DimensionDenotation.md) (such as "DATA_BATCH," or a custom denotation).  Then when creating the session, specify the dimension size for each denotation.  This can be done using the OnnxRuntime *AddFreeDimensionOverride* ABI.
+1. Constant folding can occur more often, there by reducing the CPU / GPU copies and stalls during evaluations.
+2. More initialization work occurs when sessions are created rather than during the first evaluation.
+3. Weights may be pre-processed within DirectML, enabling more efficient algorithms.
+4. Graph optimization occurs within DirectML. For example, Concat operators may be removed and more optimal tensor layouts may be used for the input and output of operators.
 
+Normally, when the shapes of model inputs are known during a session creation, the shapes for the rest of the model are inferred by OnnxRuntime when that session is created.  
+
+However, if a model input contains a free dimension (such as for batch size), additional steps must be taken to retain the above performance benefits. These include:
+
+1. Edit the model to replace an input's free dimension (specified through ONNX using "dim_param") with a fixed size (specified through ONNX using "dim_value").
+2. Specify values of named dimensions within model inputs when creating the session using the OnnxRuntime *AddFreeDimensionOverrideByName* ABI.
+3. Edit the model to ensure that an input's free dimension has a [denotation](https://github.com/onnx/onnx/blob/master/docs/DimensionDenotation.md) (such as "DATA_BATCH," or a custom denotation).  Then, when creating the session, specify the dimension size for each denotation. This can be done using the OnnxRuntime *AddFreeDimensionOverride* ABI.
+
+<p><a href="#">Back to top</a></p>
 
 ## Samples
 
+DirectML and Onnx Runtime can be used to achieve many models. Here are some examples given on [Github](https://github.com/microsoft/DirectML#directml-samples):
```

Comment:
Copying the list of mdoels from that page to this one seems worse to me than just leaving a link here.